src/devices/tuya.ts

```diff
@@ -5310,6 +5310,165 @@ const definitions: Definition[] = [
+                    Normal: Orange while off and white while on.
+                    On: Always white. Off: Always orange.
+                    Flash: Flashes white when triggered.
+                    Note: Orange light will turn off after light off delay, white light always stays on'`),
+            new exposes.Enum('light_mode_l5', ea.STATE_SET, ['normal', 'on', 'off', 'flash']).
+                withDescription(`'Sets the indicator mode of l5.
+                    Normal: Orange while off and white while on.
+                    On: Always white. Off: Always orange.
+                    Flash: Flashes white when triggered.
+                    Note: Orange light will turn off after light off delay, white light always stays on'`),
+            new exposes.Enum('light_mode_l6', ea.STATE_SET, ['normal', 'on', 'off', 'flash']).
+                withDescription(`'Sets the indicator mode of l6.
+                    Normal: Orange while off and white while on.
+                    On: Always white. Off: Always orange.
+                    Flash: Flashes white when triggered.
+                    Note: Orange light will turn off after light off delay, white light always stays on'`),
+            new exposes.Enum('light_mode_l7', ea.STATE_SET, ['normal', 'on', 'off', 'flash']).
+                withDescription(`'Sets the indicator mode of l7.
+                    Normal: Orange while off and white while on.
+                    On: Always white. Off: Always orange.
+                    Flash: Flashes white when triggered.
+                    Note: Orange light will turn off after light off delay, white light always stays on'`),
+            new exposes.Enum('light_mode_l8', ea.STATE_SET, ['normal', 'on', 'off', 'flash']).
+                withDescription(`'Sets the indicator mode of l8.
+                    Normal: Orange while off and white while on.
+                    On: Always white. Off: Always orange.
+                    Flash: Flashes white when triggered.
+                    Note: Orange light will turn off after light off delay, white light always stays on'`),
+        ],
+        endpoint: (device) => {
+            return {'l1': 1, 'l2': 1, 'l3': 1, 'l4': 1, 'l5': 1, 'l6': 1, 'state': 1, 'backlight': 1, 'l7': 1, 'l8': 1};
+        },
+        meta: {
+            multiEndpoint: true,
+            tuyaDatapoints: [
+                [1, 'state_l1', tuya.valueConverter.onOff],
+                [2, 'state_l2', tuya.valueConverter.onOff],
+                [3, 'state_l3', tuya.valueConverter.onOff],
+                [4, 'state_l4', tuya.valueConverter.onOff],
+                [5, 'state_l5', tuya.valueConverter.onOff],
+                [6, 'state_l6', tuya.valueConverter.onOff],
+                [16, 'backlight_mode', tuya.valueConverter.onOff],
+                [29, 'power_on_behavior_l1', tuya.valueConverterBasic.lookup({'off': tuya.enum(0), 'on': tuya.enum(1), 'memory': tuya.enum(2)})],
```

Comment:
Please use `tuya.valueConverter.powerOnBehavior` here

Reply:
using `tuya.valueConverter.powerOnBehavior` results in the following error
```
Zigbee2MQTT:error 2023-07-14 10:08:38: Publish 'set' 'power_on_behavior' to '0xe0798dfffe66bfb1' failed: 'TypeError: Cannot read properties of undefined (reading 'to')'
Zigbee2MQTT:debug 2023-07-14 10:08:38: TypeError: Cannot read properties of undefined (reading 'to')
    at Object.convertSet (E:\zigbee2mqtt\zigbee2mqtt\node_modules\zigbee-herdsman-converters\src\lib\tuya.js:1447:54)
    at Publish.onMQTTMessage (E:\zigbee2mqtt\zigbee2mqtt\lib\extension\publish.ts:247:52)
    at EventEmitter.emit (node:events:525:35)
    at EventBus.emitMQTTMessage (E:\zigbee2mqtt\zigbee2mqtt\lib\eventBus.ts:109:22)
    at MQTT.onMessage (E:\zigbee2mqtt\zigbee2mqtt\lib\mqtt.ts:140:27)
    at WebSocket.<anonymous> (E:\zigbee2mqtt\zigbee2mqtt\lib\extension\frontend.ts:126:27)
    at WebSocket.emit (node:events:513:28)
    at Receiver.receiverOnMessage (E:\zigbee2mqtt\zigbee2mqtt\node_modules\ws\lib\websocket.js:1184:20)      
    at Receiver.emit (node:events:513:28)
    at Receiver.dataMessage (E:\zigbee2mqtt\zigbee2mqtt\node_modules\ws\lib\receiver.js:541:14)
```

Reply:
The four previous devices I've added also used
`[29, 'power_on_behavior_l1', tuya.valueConverterBasic.lookup({'off': tuya.enum(0), 'on': tuya.enum(1), 'memory': tuya.enum(2)})],`
Should I create a new converter in `tuya.valueConverter` for this kind of `power_on_behavior`

Reply:
Yes, but please use `previous` instead of `memory` (since most other devices also use `previous`)

Reply:
Changed